asab/zookeeper/container.py

```diff
@@ -28,25 +28,32 @@ def __init__(self, app, config_section_name, config=None):
 		self.ZooKeeper = aiozk.ZKClient(self.Config["servers"])
 		self.ZooKeeperPath = self.Config["path"]
 
+
 	async def initialize(self, app):
 		await self.ZooKeeper.start()
 		await self.ZooKeeper.ensure_path(self.ZooKeeperPath)
 
 	async def finalize(self, app):
 		await self.ZooKeeper.close()
 
-	async def advertise(self, data, path, encoding="utf-8"):
-		if isinstance(data, dict):
-			data = json.dumps(data).encode(encoding)
-		elif isinstance(data, str):
-			data = data.encode(encoding)
-		elif asyncio.iscoroutinefunction(data):
-			data = await data
-		elif callable(data):
-			data = data()
+	async def advertise(self,data, path):
+		self.Data =data
+		self.Path = path
+		await self.on_tick()
+		self.App.PubSub.subscribe("Application.tick/300!", self.on_tick)
+
+	async def on_tick(self,encoding="utf-8"):
+		if isinstance(self.Data, dict):
```

Comment:
Well, the actual functionality should be in a method called `_do_advertise()` ... which is called in initializator and on tick.